thoth/storages/graph/postgres.py

```diff
@@ -5001,6 +5001,48 @@ def sync_dependency_monkey_result(self, document: dict) -> None:
 
                 inspection_run.dependency_monkey_run_id = dependency_monkey_run.id
 
+    def get_python_package_required_symbols(
+        self, package_name: str, package_version: str, index_url: str
+    ) -> List[str]:
```

Comment:
Using `query.first()` you can have None indeed, therefore the output should be `Union[List[str], str]` because it can have None.

Reply:
and we convert None type to a 'None' string?

Reply:
+1 Indeed it can be `Union[List[str], None]` or better `Optional[List[str]]`. Thanks :)

Reply:
I think the signature should define return type of `List[str]` - if there are no symbols, an empty list should be returned.

IMHO `first` returns [the very first result of the query](https://docs.sqlalchemy.org/en/13/orm/query.html). We want to retrieve all symbols, not just the first one.

Reply:
But there is no `first` right now anymore, right?

Reply:
Here we need to add `_all` suffix as well instead :)

Reply:
Yes, Kevin addressed my earlier comments already. The return value IMHO should be `List[str]`

Reply:
Yes, the comments above, where related to previous implementation when it was only one simbol and if not present was giving `None`.